File: Python/Programmers/Level 1/하샤드 수.py
# def solution(arr):
#     total = sum([int(num) for num in str(arr)]) # 각 자리수 합
#     return arr % total == 0

def solution(x):
    return bool(not x % sum(map(int, str(x)))) # sum, map 함수를 이용하여 각 자리수의 합을 구하고, bool 함수를 이용하여 boolean 값으로 변환

File: JavaScript/Programmers/Level 1/하샤드 수.js
function solution(x) {
    // x를 문자열로 변환 후 배열로 변환, 이후 reduce 메소드를 이용하여 x의 각 자리의 합을 구함
    const num = [...(x + "")].reduce((sum, c) => sum + +c, 0);
    return !(x % num); // 나누어 떨어진다면 하샤드 수
}
